src/components/ProductInfo/ProductInfo.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { fetchProductById } from '../../features/productSlice';
+import { Loader } from '../Loader';
+import styles from './ProductInfo.module.scss';
+
+export const ProductInfo: React.FC = () => {
+  const { productId = '' } = useParams();
+  const dispatch = useAppDispatch();
+  const { product, isLoading, error } = useAppSelector(
+    (state) => state.product
+  );
+
+  const propertiesData = [
+    {
+      name: 'Weight (imperial)',
+      value: product?.weight.imperial,
+    },
+    {
+      name: 'Weight (metric)',
+      value: product?.weight.metric,
```

Comment:
use optional chaining to get deeply nested properties like product?.weight.metric or product?.height.imperial